File: dsa/array_string_obj/reverseArray.js
function reverse(arr) {
  let i = 0, j = arr.length - 1;
  while (i < j) {
    [arr[i], arr[j]] = [arr[j], arr[i]];
    i++; j--;
  }
  return arr;
}

console.log(reverse([1, 2, 3, 4])); // [4,3,2,1]

File: dsa/array_string_obj/reverseArray.py
def reverse_array(arr):
    n = len(arr)
    i, j = 0, n - 1
    while i < j:
        arr[i], arr[j] = arr[j], arr[i]
        i += 1
        j -= 1
    return arr

print(reverse_array([2, 4, 7, 8, 9]))
